docs: rename RedundantAssignEliminator to UnusedAssignEliminator + other small corrections (#14454)
ethereum/solidity: docs/installing-solidity.rst

    docker run ethereum/solc:stable --help

> You can specify release build versions in the tag. For example,

Reviewer: ```suggestion
You can specify release build versions in the tag. For example:
```

.. code-block:: bash

>     docker run ethereum/solc:0.8.21 --help

Reviewer: The comment in the paragraph above and code-block in question with the version `0.5.4` was only meant as an example of a specific docker image tag. I believe we don't plan to keep bumping this information on every release, so I don't really see need to update it. But I'm fine with the change if you think it makes it more clear. The typo `For example, You...` should be fixed indeed.

Author: >I believe we don't plan to keep bumping this information on every release

Yes, I agree with you! But do you think the version 0.5.4 is a too old?

Btw, I saw the following tweet about the Vyper bug.

https://twitter.com/WuBlockchain/status/1685827572033363968

>Vyper official documentation recommendation is actually a faulty version. A bug in the smart contract language layer means that almost all protocols using Vyper are affected. Fortunately most protocols such as Uniswap use Solidity instead of the less popular Vyper. According to Slowmist 

This tweet is wrong. It's clear from the Vyper's documentation that the 0.2.15 version is just an example version, and they don't recommend the 0.2.15 version.
But when I see this kind of misunderstanding, I think the version should be updated once in a while.

Reviewer: > > I believe we don't plan to keep bumping this information on every release
> 
> Yes, I agree with you! But do you think the version 0.5.4 is a too old?
> 

Yes it is, but as it is mean as an example, I don't really expect that anyone would think that we are recommending such version, because we are not. However, I'm fine with the change, but this version will become obsolete in the next release anyway.

Reviewer 2: Like @r0qs said, this was not meant to be kept up to date and will go out of date again soon. But you do have a point. Some users could copy it without making sure it's a recent version, especially if they're new to Solidity and don't know that 0.5.x is ancient by now. We should replace it with a tag pointing at the latest version and just mention in the surrounding text that tags for specific versions are available too. Then an out of date version would not be such a problem because people would get the latest one by default.

```suggestion
    docker run ethereum/solc:stable --help
```

Reviewer 2: Also, this currently sits in a commit described as `fix: typo`. Something like this should really get a separate commit.

Author: @cameel Thanks for your comment.
The current documentation already describes a command using `stable` as follows, and changing the specific version to `stable` seems unnatural to me.
![image](https://github.com/ethereum/solidity/assets/20497787/4d3950b8-947e-40ae-95a2-409a6ce415cc)

As an alternative, I think using `0.x.y` is good: `docker run ethereum/solc:0.x.y --help`.
`0.x.y` is used at the beginning of this page and on other pages, so users can understand it.

![image](https://github.com/ethereum/solidity/assets/20497787/333ddb21-a0ca-4bd5-a9a0-24dfec532fa4)

You're right about the commit issue, so I split it into separate commits!

Reviewer 2: Ah, true :) I did not notice that we already have `stable` above.

In that case I don't think this bit with a specific version is even needed. I mean, with a concrete version it did make some sense, but if you replace it with `x.y.z`, I think we're better off only mentioning that as a part of the surrounding text like I suggested.

BTW, I'd use `x.y.z` instead of `0.x.z`. That bit you screenshotted specifically refers to major release `0` so it's relevant there, but in the rest of the docs, we don't have to assume we're at `0`. The number will go up some day and the less we'll have to update then the better :)

Reviewer: > In that case I don't think this bit with a specific version is even needed. I mean, with a concrete version it did make some sense, but if you replace it with `x.y.z`, I think we're better off only mentioning that as a part of the surrounding text like I suggested.

Yeah, maybe it is better indeed to do what @cameel suggested and only mention the use with tags for specific version in the surrounding text, removing it from the example. @minaminao if you do that, we can already merge it :)

To use the Docker image to compile Solidity files on the host machine, mount a

> local folder for input and output, and specify the contract to compile. For example,

Reviewer: I would use colon here:
```suggestion
local folder for input and output, and specify the contract to compile. For example:
```

    docker run ethereum/solc:stable --help

You can specify release build versions in the tag. For example:

.. code-block:: bash

    docker run ethereum/solc:stable --help

Note

> Specific version tags are supported for the Docker tag such as `docker run ethereum/solc:0.8.23 --help`. We will be passing the

Reviewer: Maybe this is sufficient? I don't think we need to repeat the whole command.

```suggestion
Specific compiler versions are supported as the Docker image tag such as `ethereum/solc:0.8.23`. We will be passing the
```

Reviewer 3: Thanks @r0qs! SGTM :)
